src/dbus/server/CMakeLists.txt

```diff
@@ -50,6 +50,7 @@ target_include_directories(otbr-dbus-server PRIVATE
 add_dependencies(otbr-dbus-server otbr-dbus-introspect-header)
 
 target_link_libraries(otbr-dbus-server PUBLIC
+    $<$<BOOL:${OTBR_MDNS}>:otbr-mdns>
```

Comment:
I wonder why this is necessary. It seems the dependency is on BorderAgent.

Reply:
This is due to the build errors introducing by border_agent.hpp which newly added in the dbus; move this check to border_agent.cpp to avoid the build errors and also unnecessary lib in dbus.